src/objective_function_filter.jl

```diff
@@ -0,0 +1,40 @@
+# Inspired from `MOI.Utilities.ModelFilter`
+
+struct ObjectiveFunctionFilter{T} <: MOI.ModelLike
+    inner::T
+    function ObjectiveFunctionFilter(model::MOI.ModelLike)
+        return new{typeof(model)}(model)
+    end
+end
+
+function MOI.get(model::ObjectiveFunctionFilter, attr::MOI.ObjectiveFunction)
+    func = MOI.get(model.inner, attr)
+    constant = MOI.constant(func)
+    if !iszero(constant)
+        func = copy(func)
+        func.constant = zero(constant)
+    end
```

Comment:
could you explain a bit more? It looks like we totally disregard the constant by throwing it away here, is that correct?

Reply:
It's saved in a field of `Optimizer` in `optimize!` and added back when the user asks for the objective value.